src/webui/www/private/scripts/client.js

```diff
@@ -579,18 +582,30 @@ window.addEvent('load', function() {
                         // re-select previously selected rows
                         torrentsTable.reselectRows(torrentsTableSelectedRows);
                 }
-                clearTimeout(syncMainDataTimer);
-                syncMainDataTimer = syncMainData.delay(getSyncMainDataInterval());
+                requestInProgress = false;
+                syncData(getSyncMainDataInterval())
+                $('syncing_div').set('text', '');
```

Comment:
And a follow-up question, can this line be moved in to `syncData()`?

Reply:
I think that would require changing the scope of `syncData` too much, no? It is called in other places without those statements surrounding it.

Reply:
@Chocobo1 to elaborate on my previous point:

there are in fact to places where the code almost repeats itself, with a pattern similar to the one you pointed out:

```js
requestInProgress = false;
syncData(getSyncMainDataInterval())
$('syncing_div').set('text', '');
```

However, this is only inside the `onSuccess` and `onFailure` logic of `syncMainData`. The `syncData` function is also called inside `updateMainData`, where it wouldn't make sense that `syncData` also toggled `requestInProgress` and the div.

Perhaps I should instead rename the variable `requestInProgress` to `syncRequestInProgress` since it is not used anywhere else and better signals the intent?

Reply:
@Chocobo1 it appears I have misunderstood your comment from the beginning. You were _only_ referring to the `$('syncing_div').set('text', '');` line, correct? In that case, yes, I think it can be moved inside.

Reply:
Addressed in latest push.